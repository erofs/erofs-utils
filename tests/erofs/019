#!/bin/sh
# SPDX-License-Identifier: GPL-2.0+
#
# 019 - check extended attribute functionality
#
seq=`basename $0`
seqres=$RESULT_DIR/$(echo $0 | awk '{print $((NF-1))"/"$NF}' FS="/")

# get standard environment, filters and checks
. "${srcdir}/common/rc"

cleanup()
{
	cd /
	rm -rf $tmp.*
}

_require_erofs
_require_xattr

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here
echo "QA output created by $seq"

if [ -z $SCRATCH_DEV ]; then
	SCRATCH_DEV=$tmp/erofs_$seq.img
	rm -f SCRATCH_DEV
fi

localdir="$tmp/$seq"
rm -rf $localdir
mkdir -p $localdir

# set random xattrs
cp -nR ../ $localdir
dirs=`ls $localdir`
for d in $dirs; do
	key=`head -20 /dev/urandom | cksum | cut -f1 -d " "`
	val="0s"`head -3 /dev/urandom | base64 -w0`
	setfattr -n user.$key -v $val $localdir/$d
done

_scratch_mkfs $localdir >> $seqres.full 2>&1 || _fail "failed to mkfs"
_scratch_mount 2>>$seqres.full

# check xattrs
_check_xattrs $localdir $SCRATCH_MNT

_scratch_unmount

echo Silence is golden
status=0
exit 0
