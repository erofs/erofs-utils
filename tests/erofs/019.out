QA output created by 019
Silence is golden
